src/Core/Extensions/DiagnosticsExtensions.cs

```diff
@@ -556,8 +556,8 @@ internal static bool IsEffective(this DiagnosticDescriptor descriptor, SyntaxNod
         SyntaxTreeOptionsProvider provider = compilationOptions.SyntaxTreeOptionsProvider;
 
         if (provider is not null
-            && !provider.TryGetGlobalDiagnosticValue(descriptor.Id, cancellationToken, out reportDiagnostic)
-            && !provider.TryGetDiagnosticValue(syntaxTree, descriptor.Id, cancellationToken, out reportDiagnostic))
+            && !provider.TryGetDiagnosticValue(syntaxTree, descriptor.Id, cancellationToken, out reportDiagnostic)
+            && !provider.TryGetGlobalDiagnosticValue(descriptor.Id, cancellationToken, out reportDiagnostic))
```

Comment:
You are right that the order is incorrect.

I think that the correct order should be file/project/global and "project-wide" configuration is stored in `compilationOptions.SpecificDiagnosticOptions`:

```
if (provider?.TryGetDiagnosticValue(syntaxTree, descriptor.Id, cancellationToken, out reportDiagnostic) != true
    && !compilationOptions.SpecificDiagnosticOptions.TryGetValue(descriptor.Id, out reportDiagnostic))
{
    provider?.TryGetGlobalDiagnosticValue(descriptor.Id, cancellationToken, out reportDiagnostic);
}
```

Also please apply similar order in the method `GetEffectiveSeverity` which is located several lines below.